### ISSUE TITLE
[A11Y] Sidebar needs aria role=menu and must get focused when opened

### ISSUE THREAD

**Issue by aghassemi** [ampproject/amphtml#5516]
We need to apply `role=menu` to `amp-sidebar` (if not defined by the author) and also when it is opened, we need to focus on it.

Currently when sidebar is opened, nothing happens, user's next tab stop is not the items inside the sidebar and nothing is announced to the user.

By doing the role and focus, TalkBack should announce something like: "Menu, navigation, list showing 10 items" (assuming there is a `<nav>` in the sidebar)

**Comment by camelburrito** [comment]
Sidebar can be use as a container for menu(s), but it itself is not a menu we do have aria hidden and shown. We should definitely set the focus on its first child.

**Comment by cramforce** [comment]
Can we enforce that the right thing is being done? Why would we not add the
role to the sidebar?

**Comment by camelburrito** [comment]
sidebar need not necessarily contain a menu. Thats why we named it sidebar.

**Comment by aghassemi** [comment]
I think default role of menu makes sense for sidebar (changed description of issue to add role=menu if not already defined by author so they can override it). 

`menu` role is a pretty generic and simply implies `widget that offers a list of choices to the user.` Also `menu` has the implicit meaning that it is an overlay that can be opened and closed. Most accessible sidebar implementation I have seen use `menu` role and so does the native android sidebar (try the hamburger menu in native Inbox app for example, it is not all just menu items, it has profile pictures, drops downs and navigation items)

**Comment by camelburrito** [comment]
that is one argument - 

The real thing that needs to be added is
`aria-controls` on the trigger. and add `aria-expanded` on the sidebar - https://www.w3.org/WAI/GL/wiki/Using_aria-expanded_to_indicate_the_state_of_a_collapsible_element

**Comment by aghassemi** [comment]
If `sidebar` did not shim and overlay the page and simply pushed the main content to the right expecting the user to go back and forth between the sidebar and main area to their task (e.g. two vs one column page layout) then I would agree with you but the current sidebar has behaviour that is very specific, e.g. it automatically closes when user interacts with the area outside of it and it overlays existing content, which are menu-like behaviour. Sidebar is not simply a collapsible container, for vast majority of cases, our sidebar is used to implement a navigation menu and what matters for a11y is what actually helps the user at the end. If native hamburger menus are announced as "menu, navigation, ...", so should ours.

**Comment by camelburrito** [comment]
@aghassemi - we might have a implementation of sidebar like that - there was some request on that and is in backlog. 

But for the existing use i guess it would not hurt to add a role - also focus seems to be more important here since i dont see a way that sidebar could be accessed without setting focus on it. Thanks for filing!
